File: FindTheJudge.java
// Time Complexity : O(n)
// Space Complexity : O(n)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this : no

class Solution {
    public int findJudge(int n, int[][] trust) {
        int[] following = new int[n + 1];
        int[] outDegree = new int[n + 1];
        
        for(int[] persons: trust) {
            following[persons[1]]++;
            following[persons[0]]--;
        }
        
        for(int i = 1; i <= n; i++) {
            if(following[i] == n-1) {
                return i;
            }
        }
        
        return -1;
    }
}

File: find_the_judge.py
# Time Complexity : O(N) where N is number of people
# Space Complexity : O(N) where N is number of people
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : No

class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        indegrees = [0 for _ in range(n)]
        
        for t in trust:
            pA, pB = t
            indegrees[pB - 1] += 1
            indegrees[pA - 1] -= 1
        
        for i in range(len(indegrees)):
            if indegrees[i] == n-1:
                return i + 1
        
        return -1